### PR TITLE
Feature/databridge

### PR DESCRIPTION
**Pull request by davtor94**

### CODE REVIEW

File: tbtaf/common/test.py [hunk 1]
```diff
@@ -4,16 +4,14 @@
 @author: S41nz
 '''
 
+from __future__ import absolute_import
 from abc import ABCMeta, abstractmethod
-from _pyio import __metaclass__
+import six
```
Comment (S41nz) [inline tbtaf/common/test.py:10]: Is this package included by default or should we document it in the README?

File: tbtaf/cx_Oracle_test.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import cx_Oracle
+import os
+import pandas as pd
+
+os.environ['TNS_ADMIN'] = '/Users/david.torres/Documents/MCAP/system-desing/TBTAF/tbtaf/adb_virt_env'
```
Comment (S41nz) [inline tbtaf/cx_Oracle_test.py:5]: Please remove any hard-coded credentials and implement logic to fetch it from OS environment variables as designed.

File: tbtaf/publisher/TBTAFPublisher.py [hunk 3]
```diff
@@ -24,23 +29,26 @@ def PublishTestPlan(self, tBTestSuiteInstance, filePath, formatFlag):
...
+            print("The file path doesn't contain a valid " + formatFlag.lower() + " file")
             raise NonSupportedFormatException("NonSupportedFormatException in PublishTestPlan")
 
         try:
-            htmlFile = open(filePath,'w+')
+            mode = 'w+' if formatFlag.lower() == "html" else 'w+b'
+            htmlFile = open(filePath, mode)
         except IOError:
-            print "Invalid filePath"
+            print("Invalid filePath")
```
Comment (S41nz) [inline tbtaf/publisher/TBTAFPublisher.py:45]: Reword as "Invalid file path"

File: tbtaf/publisher/TBTAFPublisher.py [hunk 4]
```diff
@@ -91,7 +99,12 @@ def PublishTestPlan(self, tBTestSuiteInstance, filePath, formatFlag):
         htmlString = htmlString.replace('r_report_time',s_reportTime)
         htmlString = htmlString.replace('r_tests',s_tests)
         
-        htmlFile.write(htmlString)
+        if formatFlag.lower() == "html":
+            htmlFile.write(htmlString)
+        else:
```
Comment (S41nz) [inline tbtaf/publisher/TBTAFPublisher.py:104]: not asumme that if !html then its gonna be PDF

File: tbtaf/publisher/TBTAFPublisher.py [hunk 5]
```diff
@@ -3,17 +3,22 @@
 
 @author: @andresmuro , @andres.alvarado , @mestradago , @rnunezc
 '''
+from __future__ import absolute_import
+from __future__ import print_function
 import datetime
 import os
+from xhtml2pdf import pisa
```
Comment (S41nz) [inline tbtaf/publisher/TBTAFPublisher.py:10]: Need to document this dependency if not part of the default Python installation if required

File: tbtaf/publisher/TBTAFPublisher.py [hunk 6]
```diff
@@ -3,17 +3,22 @@
 
 @author: @andresmuro , @andres.alvarado , @mestradago , @rnunezc
 '''
+from __future__ import absolute_import
+from __future__ import print_function
```
Comment (S41nz) [inline tbtaf/publisher/TBTAFPublisher.py:7]: In general, one  request on the design dimension. As you see all the bifucations on the implementation you can tell where all this is going. Therefore I request to implement a strategy pattern here in order to isolate the implementation specifics from TBTAFPublisher.

This way, TBTAFPublisher just calls a factory method which delivers a TBTAFReportGenerator from which we could have concrete implementations for HTML and PDF. This way TBTAF will just call the abstract methods but the specifics remain hidden at the concrete TBTAFReportGenerator implementation.

File: tbtaf/databridge/TBTAFDataBridge.py [hunk 7]
```diff
@@ -0,0 +1,178 @@
+from __future__ import absolute_import
+import cx_Oracle
+import os
+from common.suite import TBTestSuite
```
Comment (S41nz) [inline tbtaf/databridge/TBTAFDataBridge.py:4]: Please use the same design approach that I mentioned for the TBTAFPublisher

File: tbtaf/cx_Oracle_test.py (hunk 2 above)
Comment (davtor94) [reply tbtaf/cx_Oracle_test.py]: I pushed this file by mistake, sorry

Removed

File: tbtaf/test/discoverer/discoverer_tester.py [hunk 8]
```diff
@@ -2,40 +2,42 @@
 Created on 20/11/2015
 '''
 
+from __future__ import absolute_import
```
Comment (S41nz) [inline tbtaf/test/discoverer/discoverer_tester.py:5]: Please add the corresponding unit tests folder and content for the Databridge and Publisher additions

**Review by S41nz — CHANGES_REQUESTED** [review]
Please check and address my one liner comments

File: tbtaf/databridge/TBTAFDataBridge.py (hunk 7 above)
Comment (davtor94) [reply tbtaf/databridge/TBTAFDataBridge.py:4]: implemented

File: tbtaf/test/discoverer/discoverer_tester.py (hunk 8 above)
Comment (davtor94) [reply tbtaf/test/discoverer/discoverer_tester.py:5]: @mgonzalez1986

File: tbtaf/common/test.py (hunk 1 above)
Comment (brody7u7) [reply tbtaf/common/test.py:10]: It's already documented on README and requirements.txt

File: tbtaf/publisher/TBTAFPublisher.py (hunk 4 above)
Comment (brody7u7) [reply tbtaf/publisher/TBTAFPublisher.py]: Fixed in commit c96db212103859741c85616c38c01e8ca5527fbd

File: tbtaf/publisher/TBTAFPublisher.py (hunk 5 above)
Comment (brody7u7) [reply tbtaf/publisher/TBTAFPublisher.py]: It's already documented on README and requirements.txt

File: tbtaf/publisher/TBTAFPublisher.py (hunk 6 above)
Comment (brody7u7) [reply tbtaf/publisher/TBTAFPublisher.py:7]: Implemented in commit c96db212103859741c85616c38c01e8ca5527fbd

File: tbtaf/publisher/TBTAFPublisher.py (hunk 3 above)
Comment (brody7u7) [reply tbtaf/publisher/TBTAFPublisher.py]: Fixed in commit 34e2ef8c44f160aa6a8191335db238ae83e27923

File: tbtaf/test/discoverer/discoverer_tester.py (hunk 8 above)
Comment (davtor94) [reply tbtaf/test/discoverer/discoverer_tester.py:5]: implemented, please re check

**Comment by davtor94** [comment]
@S41nz all the comments had been addressed, please help us re checking the code

**Review by S41nz — APPROVED** [review]
Thanks team

**Pull request merged**